scripts/pure-component-fallback.js

```diff
@@ -0,0 +1,67 @@
+module.exports = function pureComponentFallback(babel) {
+  const { types: t, template } = babel;
+
+  const buildPureOrNormalSuperclass = () => t.LogicalExpression('||',
+    t.memberExpression(t.identifier('React'), t.identifier('PureComponent')),
+    t.memberExpression(t.identifier('React'), t.identifier('Component')));
+
+  const buildShouldComponentUpdate = () => {
+    const method = t.classMethod(
+      'method',
+      t.identifier('shouldComponentUpdate'),
+      [t.identifier('nextProps'), t.identifier('nextState')],
+      t.blockStatement([template('return shallowCompare(this, nextProps, nextState);')()]),
+    );
+    return method;
+  };
+
+  function superclassIsPureComponent(path) {
+    const superclass = path.get('superClass');
+    // check for PureComponent
+    if (t.isIdentifier(superclass)) {
+      return superclass.node.name === 'PureComponent';
```

Comment:
probably fine for "just this repo", but if we'll want to use this in other projects, we should make sure that `PureComponent` is actually a named import from `'react'` or a destructure from React imported/required from `'react'`